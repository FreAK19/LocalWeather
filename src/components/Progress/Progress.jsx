import React from 'react';
import CircularProgress from 'material-ui/CircularProgress/CircularProgress';
import PropTypes from 'prop-types';
import './Progress.less';

const colorSpinner = '#485';

const Progress = ({ show }) =>
  show ? (
    <div className="spinner">
      <div className="spinner__box">
        <CircularProgress size={60} color={colorSpinner} />
      </div>
    </div>
  ) : null;

Progress.propTypes = { show: PropTypes.bool.isRequired };

export default Progress;
