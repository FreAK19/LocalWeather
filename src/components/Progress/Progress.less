@bg-color: rgba(0, 188, 212, 0.48);

.spinner {
  position: absolute;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  background: @bg-color;
  z-index: 999;

  &__box {
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
  }
}