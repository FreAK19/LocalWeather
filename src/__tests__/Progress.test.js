import React from 'react';
import Progress from '../components/Progress/Progress'

test('Progress should render correctly', () => {
  const component = shallow(<Progress show/>);
  expect(component).toMatchSnapshot();
});

test('If show equals to false should render nothing', () => {
  const component = shallow(<Progress show={false}/>);
  expect(component.children().length).toBe(0);
});