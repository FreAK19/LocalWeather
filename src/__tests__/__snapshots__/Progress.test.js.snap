// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress should render correctly 1`] = `
<div
  className="spinner"
>
  <div
    className="spinner__box"
  >
    <CircularProgress
      color="#485"
      max={100}
      min={0}
      mode="indeterminate"
      size={60}
      thickness={3.5}
      value={0}
    />
  </div>
</div>
`;
